packages/components/src/popover/index.tsx

```diff
@@ -287,11 +287,45 @@ const UnforwardedPopover = (
 				? undefined
 				: normalizedPlacementFromProps,
 		middleware,
-		whileElementsMounted: ( referenceParam, floatingParam, updateParam ) =>
-			autoUpdate( referenceParam, floatingParam, updateParam, {
-				layoutShift: false,
-				animationFrame: true,
-			} ),
+		whileElementsMounted: (
+			referenceParam,
+			floatingParam,
+			updateParam
+		) => {
+			const cleanupAutoUpdate = autoUpdate(
+				referenceParam,
+				floatingParam,
+				updateParam,
+				{
+					layoutShift: false,
+					animationFrame: true,
+				}
+			);
+
+			// Observe mutations on the parent div of the inserter to update toolbar position.
+			const targetElement = document.querySelector(
+				'.interface-interface-skeleton__body'
```

Comment:
The `Popover` component is part of `@wordpress/components`'s API. `interface-interface-skeleton__body` is declared within `InterfaceSkeleton`, which is part of `@wordpress/interface`. 

However, `@wordpress/components` does not know about `@wordpress/interface` at all, nor it should care about the inserter, so tracking mutations of an unknown, unrelated to `Popover` element doesn't make much sense. 

If this is the right fix, it likely needs to be part of `InterfaceSkeleton`'s implementation, or the inserter, and not part of `Popover`'s internals. 

I'm not convinced this is the right fix, anyway. This looks like something that should be resolved within `floating-ui`, cc @ciampo and @jsnajdr who know its internals better than me.

Reply:
Yup, Marin is right. The `@wordpress/components` package is not aware of the `@wordpress/interface`, and can't be tightly coupled to any of its references.

We should look for another way to fix this.

My gut feeling tells me that the bug is currently caused by the fact that the popover anchor updates are not tracked and used to re-render the popover.